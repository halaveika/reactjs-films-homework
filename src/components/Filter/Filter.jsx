import React from 'react';
import { PageHeader} from 'antd';
import FilterNavBar from '../FilterNavBar';
import PropTypes from 'prop-types';
import ColumnsBtn from '../ColumnsBtn'
import RowsBtn from '../RowsBtn';
import './Filter.scss';

export default function Filter({handleColumn, handlRow,genres,getTrending,getTopRated, getUpcoming,handleGenre,activeGenre,setFilter,filter}) {
  return (
    <PageHeader ghost className="site-page-header">
      <FilterNavBar
        genres={genres}
        getTrending={getTrending}
        getTopRated={getTopRated}
        getUpcoming={getUpcoming}
        handleGenre={handleGenre}
        activeGenre={activeGenre}
        setFilter={setFilter}
        filter={filter}
        >
      </FilterNavBar>
      <div className="btn-container">
        <RowsBtn handlRow={handlRow}></RowsBtn>
        <ColumnsBtn handleColumn={handleColumn}></ColumnsBtn>
      </div>
    </PageHeader>
  );
}


Filter.propTypes = {
  handlRow: PropTypes.func.isRequired,
  handleColumn: PropTypes.func.isRequired,
  genres: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string})),
  getTrending: PropTypes.func.isRequired,
  getTopRated: PropTypes.func.isRequired,
  getUpcoming: PropTypes.func.isRequired,
  handleGenre: PropTypes.func.isRequired,
  activeGenre: PropTypes.string.isRequired,
  setFilter: PropTypes.func.isRequired,
  filter: PropTypes.string.isRequired,
};
