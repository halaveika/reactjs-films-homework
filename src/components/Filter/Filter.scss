.site-page-header {
  max-width: 1200px;
  width: 100%;
  border: 1px solid rgb(235, 237, 240);
  padding: 0 75px 0 75px;
  margin: 0 auto;
}

.ant-page-header-content {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding: 0;
}

.btn-container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 45px;
  height: 100%;
}
