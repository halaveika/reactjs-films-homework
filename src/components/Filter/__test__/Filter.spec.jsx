import React from 'react';
import { create } from 'react-test-renderer';
import Filter from '..';

const mockFilterProps = {
  genres: [{
    id: 13,
    name: 'Adventure',
  }, {
    id: 18,
    name: 'Drama',
  }, {
    id: 20,
    name: 'Family',
  }, {
    id: 21,
    name: 'Fantasy',
  }],
  filter: 'TRENDING',
  getContent: jest.fn(),
  GetGenres: jest.fn(),
  setFilter: jest.fn(),
  setCurrentPage: jest.fn(),
  setGenre: jest.fn(),
  setListFlexDirection: jest.fn()
};

describe('test Filter component', () => {

  it('should render Filter component', () => {
    const component = create(
          <Filter
            genres={mockFilterProps.genres}
            setGenre={mockFilterProps.setGenre}
            activeGenre={mockFilterProps.activeGenre}
            setFilter={mockFilterProps.setFilter}
            filter={mockFilterProps.filter}
            setCurrentPage={mockFilterProps.setCurrentPage}
            setListFlexDirection={mockFilterProps.setListFlexDirection}
          />,
      );
     const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
