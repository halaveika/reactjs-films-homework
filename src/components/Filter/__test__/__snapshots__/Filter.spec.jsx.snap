// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Filter component should render Filter component 1`] = `
<div
  className="ant-page-header site-page-header ant-page-header-ghost"
>
  <div
    className="ant-page-header-content"
  >
    <div
      className="filter-navbar"
    >
      <div
        className="ant-tabs ant-tabs-top"
      >
        <div
          className="ant-tabs-nav"
          onKeyDown={[Function]}
          role="tablist"
        >
          <div
            className="ant-tabs-nav-wrap"
          >
            <div
              className="ant-tabs-nav-list"
              style={
                Object {
                  "transform": "translate(0px, 0px)",
                  "transition": undefined,
                }
              }
            >
              <div
                className="ant-tabs-tab ant-tabs-tab-active"
                onClick={[Function]}
              >
                <div
                  aria-controls={null}
                  aria-selected={true}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  Trending
                </div>
              </div>
              <div
                className="ant-tabs-tab"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": undefined,
                  }
                }
              >
                <div
                  aria-controls={null}
                  aria-selected={false}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  Top Rated
                </div>
              </div>
              <div
                className="ant-tabs-tab"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": undefined,
                  }
                }
              >
                <div
                  aria-controls={null}
                  aria-selected={false}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  Coming Soon
                </div>
              </div>
              <div
                className="ant-tabs-tab"
                onClick={[Function]}
                style={
                  Object {
                    "marginLeft": undefined,
                  }
                }
              >
                <div
                  aria-controls={null}
                  aria-selected={false}
                  className="ant-tabs-tab-btn"
                  id={null}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  role="tab"
                  tabIndex={0}
                >
                  <span
                    className="ant-dropdown-trigger ant-dropdown-link"
                    onClick={[Function]}
                    onMouseDown={[Function]}
                    onTouchStart={[Function]}
                    placement="bottomCenter"
                  >
                    Genre 
                    <span
                      aria-label="down"
                      className="anticon anticon-down"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="down"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                        />
                      </svg>
                    </span>
                  </span>
                  <div />
                </div>
              </div>
              <div
                className="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
              />
            </div>
          </div>
          <div
            className="ant-tabs-nav-operations ant-tabs-nav-operations-hidden"
          >
            <button
              aria-controls="null-more-popup"
              aria-expanded={false}
              aria-haspopup="listbox"
              aria-hidden="true"
              className="ant-tabs-nav-more"
              id="null-more"
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "marginLeft": undefined,
                  "order": 1,
                  "visibility": "hidden",
                }
              }
              tabIndex={-1}
              type="button"
            >
              <span
                aria-label="ellipsis"
                className="anticon anticon-ellipsis"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="ellipsis"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
        <div
          className="ant-tabs-content-holder"
        >
          <div
            className="ant-tabs-content ant-tabs-content-top"
            style={null}
          >
            <div
              aria-hidden={false}
              aria-labelledby={null}
              className="ant-tabs-tabpane ant-tabs-tabpane-active"
              id={null}
              role="tabpanel"
              style={Object {}}
              tabIndex={0}
            />
            <div
              aria-hidden={true}
              aria-labelledby={null}
              className="ant-tabs-tabpane"
              id={null}
              role="tabpanel"
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
            />
            <div
              aria-hidden={true}
              aria-labelledby={null}
              className="ant-tabs-tabpane"
              id={null}
              role="tabpanel"
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
            />
            <div
              aria-hidden={true}
              aria-labelledby={null}
              className="ant-tabs-tabpane"
              id={null}
              role="tabpanel"
              style={
                Object {
                  "display": "none",
                }
              }
              tabIndex={-1}
            />
          </div>
        </div>
      </div>
      <span
        className="genre-status"
      />
    </div>
    <div
      className="btn-container"
    >
      <div
        className="rows-btn"
        onClick={[Function]}
      >
        <div />
        <div />
        <div />
        <div />
      </div>
      <div
        className="columns-btn"
        onClick={[Function]}
      >
        <div />
        <div />
      </div>
    </div>
  </div>
</div>
`;
